### PR TITLE
Integration gateway improvements

### LINKED ISSUE #517: Integration gateway response postfilter [PlanID: 2023.3/45/1.7]
Currently there is only support for incoming requests to integration gateway, there needs to be support for response as well, especially coming from workflows.

There's an existing JSON postfilter that should be extended/replaced with the new `Response` postfilter.

It need to effectively:
 - fetch data either from context or from workflows expressions
 - feed expression into the configuration
 - output data as needed with various http headers
 
**Frontend tasks:**
- [x] header list on modal to add multiple headers to output
- [x] input (this will be an input for expr values, similar as to what workflows have)

**Backend tasks:**
- [x] extend JSON postfilter with new Response filter implementation
- [x] tests
- [ ] documentation on various possible responses wf->apigw

The payload example for postfilter configuration on the frontend:
```
{
    "headers": {
        "content-type": [
            "text/json"
        ]
    },
    "content": "{\"expr\": \"toJSON(records)\", \"type\": \"String\"}"
}
```

### LINKED ISSUE #521: Add global integration gateway profiler setting
Triggering of global/filter profiler should be in settings, not only as env variable

Checklist:

- [x] update service to read from settings
- [x] add settings to gateway page UI
- [ ] add a new batch of cypress tests with the global profiler ON

### LINKED ISSUE #522: Integration gateway UI cleanup [PlanID: 2023.3/45/1.7]
## Route details

**Frontend**

- [x] add description field

**Backend**

- [x] add description field

## Route list

**Frontend**

- [x] [route list consistency](https://github.com/cortezaproject/corteza/issues/518)
- [x] [sort apigw filters](https://github.com/cortezaproject/corteza/issues/519)
- [x] lower-case the `Deleted routes` filter label

## Profiler

**Frontend**
- [x] add padding below body on profiler hit details (this is a general layout foo on admin, so it should be fixed by fixing that)
- [x] change edit icon on aggregations/route list to something more informative
- [x] add text description on alt on (edit) icons
- [x] add bytes to Content Length on hit details
- [x] [implement purging of old profiler hits](https://github.com/cortezaproject/corteza/issues/520)

**Backend**
- [x] [implement purging of old profiler hits](https://github.com/cortezaproject/corteza/issues/520)
- [x] fix "load more" info on 0 remaining items
- [x] [add global integration gateway profiler setting](https://github.com/cortezaproject/corteza/issues/521)

### PR DESCRIPTION
**Pull request by petergrlica**
# The following changes are implemented
- new postfilter in integration gateway
- settings support in gateway service (also in opt)
- purge endpoints
- profiler paging fix

Issues: https://github.com/cortezaproject/corteza/issues/517, https://github.com/cortezaproject/corteza/issues/522

# Changes in the user interface:
- postfilter expressions and header UI added
- filter sorting (drag drop)
- various UI fixes

Issues: https://github.com/cortezaproject/corteza/issues/522

# Checklist when submitting a final (!draft) PR
 - [x] Commits are tidied up, squashed if needed and follow guidelines in CONTRIBUTING.md
 - [x] Code builds
 - [x] All existing tests pass
 - [x] All new critical code is covered by tests
 - [x] PR is linked to the relevant issue(s)
 - [x] Rebased with the target branch

### CODE REVIEW

File: server/pkg/apigw/filter/postfilter.go [hunk 1]
```diff
@@ -274,16 +286,23 @@ func (j jsonResponse) Handler() types.HandlerFunc {
...
-			e := json.NewEncoder(rw)
-			err = e.Encode(v)
+				if h == "Content-Type" || h == "content-type" {
+					hasJsonHeader = vv == "application/json"
+				}
+			}
 		}
 
+		if reflect.ValueOf(evald).Kind() != reflect.String && hasJsonHeader {
+			err = (json.NewEncoder(rw)).Encode(expr.UntypedValue(evald))
```
Comment (tjerman) [inline server/pkg/apigw/filter/postfilter.go:301]: Have you tested if this works fine with records, modules, and similar types? We had (have) some issues with json encoding those

File: client/web/admin/src/components/Apigw/Profiler/CProfilerRouteHits.vue [hunk 2]
```diff
@@ -13,22 +13,33 @@
...
+
+        <c-input-confirm
+          :disabled="!items.length"
+          :borderless="false"
+          variant="danger"
+          @confirmed="purgeRequests"
         >
-          {{ $t('general:label.refresh') }}
-        </b-button>
+          {{ $t('purge.this') }}
```
Comment (tjerman) [inline client/web/admin/src/components/Apigw/Profiler/CProfilerRouteHits.vue:41]: `.this` ?

**Review by tjerman — APPROVED** [review]
lgtm; see 2 comments

File: server/pkg/apigw/filter/postfilter.go (hunk 1 above)
Comment (petergrlica) [reply server/pkg/apigw/filter/postfilter.go:301]: Tested yes, thanks for the observation

File: client/web/admin/src/components/Apigw/Profiler/CProfilerRouteHits.vue (hunk 2 above)
Comment (Fajfa) [reply client/web/admin/src/components/Apigw/Profiler/CProfilerRouteHits.vue:41]: Yes this route,

**Pull request merged**